### ISSUE TITLE
comment_references rule false-positive on markdown code blocks.

### ISSUE THREAD

**Issue by dikmax** [dart-lang/linter#295]
Consider this example:

``````dart
/// Some description
///
/// ```dart
/// /**
///  * Throws a [StateError] if ...
///  * similar to [anotherMethod], but ...
///  */
/// ```
///
/// More text.
void method() {}
``````

In that case `[StateError]` and `[anotherMethod]` are not references.

**Comment by pq** [comment]
@scheglov : does analyzer do the right thing here?  I'm guessing it's trying to resolve `StateError` and `anotherMethod` as if they were references.

**Comment by bwilkerson** [comment]
I suspect that's true. I don't think our parser is very intelligent about handling code blocks.

**Comment by scheglov** [comment]
Yes, we handle single line GitHub style \`code blocks\`, but we don't handle multi-line GitHub style comments.

**Comment by pq** [comment]
> Yes, we handle single line GitHub style `code blocks`, but we don't handle multi-line GitHub style comments.

Should we?  Does `dartdoc`?  (cc @keertip)

**Comment by scheglov** [comment]
https://codereview.chromium.org/2302003003

**Comment by scheglov** [comment]
https://github.com/dart-lang/sdk/commit/75d1ff092e24c1fc25824b415d637b856ce4a666
